utils/MyCurator.py

```diff
@@ -271,173 +398,144 @@ def process_generic_roi(self, roi):
-                    acquisition_label,
-                ) = self.get_file_hierarchy(file)
-
-                file_name = Path(file.name)
-
-                suffix = "".join(file_name.suffixes[-2:])
-                file_type = KNOWN_EXTENSIONS.get(suffix, suffix[1:])
+                log.debug("FOUND")
 
                 if pk == "roi":
                     namespace = file.info.get(pk, {})
-                    temp_dict = self.process_namespace_roi(
+                    output_dict = self.process_namespace_roi(
                         namespace,
                         output_dict,
-                        group_label,
-                        project_label,
-                        subject_label,
-                        session_label,
-                        acquisition_label,
-                        file_name,
-                        file_type,
+                        file
                     )
 
-                    for d in temp_dict:
-                        if d in output_dict:
-                            output_dict[d].extend(temp_dict[d])
-                        else:
-                            output_dict[d] = temp_dict[d]
+                    # for d in temp_dict:
+                    #     if d in output_dict:
+                    #         output_dict[d].extend(temp_dict[d])
+                    #     else:
+                    #         output_dict[d] = temp_dict[d]
 
                 elif pk == "ohifViewer":
                     namespace = file.info.get(pk, {}).get("measurements", {})
-                    temp_dict = self.process_namespace_ohifViewer(
-                        namespace,
-                        output_dict,
-                        group_label,
-                        project_label,
-                        subject_label,
-                        session_label,
-                        acquisition_label,
-                        file_name,
-                        file_type,
-                    )
-
-                    for d in temp_dict:
-                        if d in output_dict:
-                            output_dict[d].extend(temp_dict[d])
-                        else:
-                            output_dict[d] = temp_dict[d]
+                    parent_ses = file.parent.parents.session
+                    if parent_ses is None:
+                        log.debug('file is not at acquisition level, skipping')
+                        continue
+                    session = self.fw.get_session(parent_ses)
+                    output_dict = self.process_namespace_ohifViewer(session, namespace, output_dict)
+
+                    # for d in temp_dict:
+                    #     if d in output_dict:
+                    #         output_dict[d].extend(temp_dict[d])
+                    #     else:
+                    #         output_dict[d] = temp_dict[d]
 
         return output_dict
 
 
+    def get_roi_dicom_file(self, fw_object, study_uid, series_uid, sop_uid):
```

Comment:
I think there might be some opportunity for an increase in efficiency here.  Just a note for the next iteration of this gear.  Since we're already walking the hierarchy, we might be able to leverage that instead of walking it again to find the file for each roi.  I'll have to think about it some more, but possibly keeping the ROIs as a dataframe on the class and checking each file for UIDs already existing on the class if we do a depth-first traversal.  This is out of scope of this MR though.

Reply:
I agree that this could probably be improved, and that it is also out of scope